src/utilities/iterators.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @module Utilities
+ * @submodule Iterators
+ * @for p5
+ * @requires core
+ */
+
+'use strict';
+
+var p5 = require('../core/core');
+
+
+/**
+ * Produces an iterator over the numbers in the range. The result is given by
+ * starting at start, and adding step repeatedly until the result has gone past
+ * stop.
+ * In the single parameter version, start is set to zero and step is set to one.
+ *
+ * @method range
+ * @param  {Number}  stop      Number to stop at
+ * @return {Iterator}          An iterator that returns the values sequentially
+ * @example
+ * <div>
+ * <code>
+ * for(var i of range(width)) {
+ *   line(i, 0, i, i);
+ * }
+ * </code>
+ * </div>
+ * <div>
+ * <code>
+ * var start = width / 4;
+ * var stop = 3 / 4 * width;
+ * for(var i of range(start, stop, 2)) {
+ *   line(i, start, i, i);
+ *   line(start, i, i, i);
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * Triangle filling the top right corner of the canvas
+ * Square in the center of the canvas, with a striped effect
+ */
+/**
+ * @method range
+ * @param  {Number}  start       Number to start at (default 0)
+ * @param  {Number}  stop
+ * @param  {Number}  [step]      Amount to step by (default 1)
+ * @return {Iterator}
+ */
+p5.prototype.range = function(start, stop, step) {
+  if(typeof start === 'undefined') {
+    throw new Error('Not enough parameters for range');
+  }
+  if(typeof stop === 'undefined') {
+    stop = start;
+    start = 0;
+  }
+  if(typeof step === 'undefined') {
+    step = 1;
+  }
+  var dir = step < 0 ? -1 : 1;
+  if(Symbol && Symbol.iterator) {
+    var res = {};
+    res[Symbol.iterator] = function() {
+      var i = start;
+      return {
+        next: function() {
+          if(dir * (i - stop) >= 0) {
+            return {
+              done: true
+            };
+          } else {
+            i += step;
+            return {
+              value: i - step,
+              done: false
+            };
+          }
+        }
+      };
+    };
+    return res;
+  } else {
+    throw new Error('range() unsuported');
```

Comment:
In this case we can return just an array of values.
It is not memory efficient, but it is a kind of acceptable polyfill

Reply:
I'd say this is a breaking polyfill - it would be very easy to crash browsers in ways that work fine as a generator:

`range(0, 1, 0);`
`range(10000000);`

etc. I'm not sure if it's good to have a polyfill. Could partial polyfill up to n numbers, though?

Reply:
I just wanted to be kind to the user,
but OK, You have convinced me ;)

Reply:
Frankly, I'm not that convinced myself...